src/Http/Http/src/ReadOnlyPipeStream.cs

```diff
@@ -11,6 +11,7 @@ namespace System.IO.Pipelines
     /// <summary>
     /// Represents a read-only Stream backed by a PipeReader
     /// </summary>
+    // REVIEW: Could this be replaced or simplified with https://apisof.net/catalog/System.IO.Pipelines.PipeReader.AsStream() ?
     public class ReadOnlyPipeStream : Stream
```

Comment:
I want to remove this. It looks like Kestrel inherits from it in `HttpRequestStream` (and theres also a `WriteOnlyPipeStream` that Kestrel uses in `HttpResponseStream`). These are public so I want to yank them if we can. I don't think Kestrel needs to use them at all though. I think it can do all the work in it's own stream.

Reply:
That sounds fine. I would do the following:
1. Move ReadOnlyPipeStream\WriteOnlyPipeStream to Kestrel project and make them internal
2. Merge HttpRequestStream with ReadOnlyPipeStream and visa versa.